sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -1,266 +1,178 @@
 ```
+### Sending events
+* `add_sender`, `run`, and `stop` methods are replaced by `from_connection_string` method on `EventGridProducerClient` to more approachably open a connection ready for sending.
+* `send` method is replaced by `send_batch` method on `EventGridProducerClient` to clarify that, instead of single `EventData`, either an `EventDataBatch` or list of `EventData` are sent in one call not exceeding the event hub frame size limit.
+* `EventDataBatch` is created using the `create_batch` method and `EventData` messages are added to the batch using the `add` method, until the size limit is reached.
 
-Becomes this in V5:
-
+In v1:
 ```python
-logger = logging.getLogger("azure.eventhub")
-async def on_event(partition_context, event):
-    logger.info("Message received:{}".format(event.body_as_str()))
-    await partition_context.update_checkpoint(event)
-
-client = EventHubConsumerClient.from_connection_string(
-    conn_str=CONNECTION_STR, consumer_group="$Default", eventhub_name=EVENTHUB_NAME
-)
-async with client:
-    await client.receive(on_event=on_event, partition_id="0", starting_position="@latest")
+    client = EventHubClient(address)
+    sender = client.add_sender()
+    client.run()
+    sender.send(EventData('Single message'))
+    client.stop()
 ```
 
-### Migrating code from `EventHubClient` and `AsyncSender` to `EventHubProducerClient` for sending events
+In v5:
+```python
+    producer_client = EventHubProducerClient.from_connection_string(conn_str, eventhub_name)
 
-In V1, you create an `EventHubClient`, then create a `AsyncSender`, and call `AsyncSender.send` to send an event that may have
-a list/generator of messages.
+    # Send EventDataBatch
+    event_data_batch = producer.create_batch()
+    event_data_batch.add(EventData('Single message'))
+    producer.send_batch(event_data_batch)
 
-In V5, this has been consolidated into a one method - `EventHubProducerClient.send_batch`.
-Batching merges information from multiple events into a single send, reducing
-the amount of network communication needed vs sending events one at a time.
-This method deterministically tells you whether the batch of events are sent to the event hub.
+    # Send list of EventData
+    event_data_batch = [EventData('Single message')]
+    producer.send_batch(event_data_batch)
+```
 
-So in V1:
+### Receiving events 
+* `add_receiver`, `run`, and `stop` methods are replaced by `from_connection_string` method on `EventGridConsumerClient` to more approachably open a connection ready for receiving.
+* `receive` method is renamed `receive_batch` on `EventGridConsumerClient` to be more consistent in the usage of `batch` suffix in other methods on the producer and consumer when receiving or sending batches.
+* `receive` method on `EventGridConsumerClient` now receives only a single event as opposed to previously receiving a batch of events to more clearly reflect the naming, in which `batch` is not used as a suffix.
```

Comment:
This suggests that the only major change is in the name of the methods where as the main change is that we now have push mode vs the previous pull mode i.e. now you have to pass callbacks, where as previously you got the events as the output of the method.